inpage/src/EthereumApi.tsx

```diff
@@ -143,10 +163,198 @@ export default class EthereumApi {
         });
       }
 
-      return (await this.#networkRequest({
-        method: 'eth_sendTransaction',
-        params: txs,
-      })) as string;
+      const account = await this.#getAccount();
+      const contracts = await this.#waxInPage.getContracts();
+
+      const actions = txs.map((tx) => {
```

Comment:
Mmm yeah it might be worth trying to get on the same page here. As far as 4337 is concerned though, technically what your account does with the calldata is out of scope of the spec, so 'action' is a term that is internal to the particular smart account implementation. I would say it should be a different term than `userOp` though, since that *is* supposed to be the 4337-level operation, not something that's interpreted internal to the smart account.

Reply:
Yeah that makes sense. I have seen the 4337 sdk has a type called `TransactionDetailsForUserOp`, used in the `ERC4337EthersSigner`. But not sure that fits any better